app/theme.tsx

```diff
@@ -12,12 +14,13 @@ interface IThemeContextProps {
 
 export const ThemeContext = React.createContext<IThemeContextProps>({ theme: 'light' });
 
-export function withTheme(Component: any): any {
-	const ThemedComponent = (props: any) => (
+export function withTheme<T extends object>(Component: React.ComponentType<T> & TNavigationOptions): typeof Component {
+	const ThemedComponent = (props: T) => (
 		<ThemeContext.Consumer>{contexts => <Component {...props} {...contexts} />}</ThemeContext.Consumer>
 	);
+	ThemedComponent.navigationOptions = Component.navigationOptions;
```

Comment:
We don't need this.
`hoitNonReactStatics`' goal is to do it.
https://reactjs.org/docs/higher-order-components.html#static-methods-must-be-copied-over

I commented it and it worked just fine.